pkg/services/ngalert/notifier/receiver_svc.go

```diff
@@ -213,6 +225,85 @@ func (rs *ReceiverService) GetReceivers(ctx context.Context, q models.GetReceive
 	return output, nil
 }
 
+// DeleteReceiver deletes a receiver by uid.
+// UID field currently does not exist, we assume the uid is a particular hashed value of the receiver name.
+func (rs *ReceiverService) DeleteReceiver(ctx context.Context, uid string, orgID int64, callerProvenance definitions.Provenance, version string) error {
+	//TODO: Check delete permissions.
+	baseCfg, err := rs.cfgStore.GetLatestAlertmanagerConfiguration(ctx, orgID)
+	if err != nil {
+		return err
+	}
+
+	cfg := definitions.PostableUserConfig{}
+	err = json.Unmarshal([]byte(baseCfg.AlertmanagerConfiguration), &cfg)
+	if err != nil {
+		return err
+	}
```

Comment:
It would be great if we could adopt this repo implementation everywhere we read\write AM config. https://github.com/grafana/grafana/blob/72182e02a4b5dbff3f0176cf85643e6aaf16b117/pkg/services/ngalert/provisioning/config.go#L29

Reply:
Agree, the dependency directions make this a bit awkward right now though since provisioning has most of the necessary models and interfaces. We'll want to reorganize and create explicit models not in the provisioning package first.